aptos-move/framework/aptos-stdlib/sources/smart_vector.move

```diff
@@ -0,0 +1,435 @@
+module aptos_framework::smart_vector {
+    use std::error;
+    use std::vector;
+    use aptos_std::big_vector::{Self, BigVector};
+    use aptos_std::math64::max;
+    use aptos_std::type_info::size_of_val;
+
+    /// Vector index is out of bounds
+    const EINDEX_OUT_OF_BOUNDS: u64 = 1;
+    /// Cannot destroy a non-empty vector
+    const EVECTOR_NOT_EMPTY: u64 = 2;
+
+    /// A Scalable vector implementation based on tables, elements are grouped into buckets with `bucket_size`.
+    /// The vector wrapping BigVector represents an option w/o `Drop` ability requirement of T to save space of the
+    /// metadata associated with BigVector when smart_vector is so small that inline_inner vector can hold all the data.
+    struct SmartVector<T> has store {
+        inline_inner: vector<T>,
+        table_inner: vector<BigVector<T>>,
```

Comment:
`first_bucket` and `extra_buckets` sound more expressive, just my 2 cents.

Reply:
I renamed them as inline_vec and big_vec